eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala

```diff
@@ -273,7 +274,21 @@ class Setup(val datadir: File,
           key
         }
       }
-      _ = if (bitcoinClient.useEclairSigner) logger.info("using eclair to sign bitcoin core transactions")
+      currentDescriptors <- bitcoinClient.listDescriptors()
+      _ <- if (bitcoinClient.useEclairSigner) {
+        logger.info("using eclair to sign bitcoin core transactions")
+        val descriptors = onchainKeyManager.getDescriptors(0, hSuffix = false).descriptors
+        if (!descriptors.forall(currentDescriptors.contains(_))) {
+          logger.info(s"current descriptors:\n$currentDescriptors")
+          logger.info(s"importing computed descriptors:\n$descriptors")
+          bitcoinClient.importDescriptors(descriptors)
```

Comment:
If we are restoring an existing eclair-backed wallet with an old timestamp this could potentially take a very long time, during which bitcoin core cannot be used.

Reply:
If we use `now` instead of the wallet's creation timestamp, then bitcoind will not rescan at all. If we are using ZMQ to watch for spent transactions it could miss some and the wallet won't know about new UTXOs that have received value while the node was offline.

Once Eclair is parsing blocks to find watched outputs then I don't think we'll need to rescan to detect spent outputs if we are replaying blocks from the period our node was down. We should probably still use a past timestamp to rescan from; for example, the current timestamp from `date +%s` or when the node went down if more than 2 hours ago. Otherwise the wallet will might lose track of some UTXOs; for example, from a cooperatively closed channel or splice out that confirmed while Eclair was down.

We should test the behavior of ZMQ and rescanning after importing a descriptor wallet to confirm it works the way I am assuming.

Reply:
Ok, having Eclair create a new wallet and import its descriptors works fine for a fresh wallet. 
If users already have an eclair-backed wallet that they want to move to a new node, they would create and import descriptors manually (public descriptors, the one they would get with  `bitcoin-cli listdescriptors`), and then update Eclair's configuration and re-start  ?

Reply:
Where eclair is set to sign for Bitcoin Core, we should check if the wallet configured in `eclair-signer.conf` exists. 
If it does exist we just start (we'll check later whether we can re-compute private keys for public key generated by Bitcoin Core).
If it doesn't exist, we create it and import descriptors, but only if the timestamp set in `eclair-signer.conf` is very recent (less then 2 hours old for example), otherwise we fail to start: if users want to restore an existing eclair-backed wallet, they should create the wallet and import descriptors manually (which may take some time) and then change Eclair's configuration and restart. WDYT ?

Reply:
Yes - that sounds right to me too.

I've updated the code in [0e4530f](https://github.com/ACINQ/eclair/pull/2685/commits/0e4530ff041bc8532284ad615e320a2770a59ca6) to import descriptors computed for the eclair signer wallet only if none are defined for the wallet, but fail if the `timestamp` defined for the eclair signer wallet is more than 2 hours old. 

If descriptors are defined, it does not check they match the computed descriptors, but Eclair will fail when it computes the first public key.

Reply:
This works but still require users to create the Eclair-backed wallet manually with the right parameters: we could instead check whether the configured wallet exists, do nothing if it does (our setup will check that it is consistent with the loaded key manager), and create it and import descriptors if it doesn't ?

Reply:
I've added some code to create the wallet in [987e727](https://github.com/ACINQ/eclair/pull/2685/commits/987e72798905a59dee3ce981b63ea8630eb56a51). There might be a more elegant way to do it, but I've tested this and it appears to work as expected.